### DISCUSSION TITLE
Streamlit App based on PyPDF

### DISCUSSION THREAD

**Discussion by SiddhantSadangi** [py-pdf/pypdf#2673] [Show and tell]
Hello,
First of all, thanks a lot for creating and maintaining PyPDF!

I recently published a Streamlit app based on PyPDF.

You can access the app here: https://pdfworkdesk.streamlit.app/
And the GitHub repo here: https://github.com/SiddhantSadangi/pdf-workdesk

I was wondering if you could add this as an example in your readme or docs, maybe?

I am also open to any feedback, and can add any links back to PyPDF as needed 🤗

**Comment by stefan6419846** [comment]
I just had a quick look at the app and repository. Could you please explain a little bit of what the Streamlit app is indeed doing.

Additionally, I recommend you to set a license for your package. Having a list of dependencies on your app usually is a good approach as well.

**Reply by SiddhantSadangi** [reply]
Thanks for the quick response, @stefan6419846 !

The app basically provides a UI for some PyPDF functions - like extraction, encryption, rotation and scaling, merging, and compression - so that it can be used by non-technical users.

The app itself has a [CC-BY-NC-SA 4.0](https://creativecommons.org/licenses/by-nc-sa/4.0/) license. Since it is not a Python package, I haven't set a licence for the source code, but I can get an MIT license if you recommend that.

The list of dependencies for the app is available in the `packages.txt` and `requirements.txt` files in the GitHub repo.

**Reply by stefan6419846** [reply]
Creative Commons licenses are rather uncommon for software and the usual recommendation is to avoid them and use software licenses like MIT, BSD-3-Clause or similar instead. I would recommend you to indeed add a license (file) to the repository itself.

Regarding the dependencies: Showing attribution in the frontend usually is nice and might even be required for some licenses and use cases (the BSD-3-Clause license requires attribution on distribution). How this might look and to which extent this will be required for you is something I am going to omit here as IANAL.

**Reply by SiddhantSadangi** [reply]
Thanks for your feedback!
Added the BSD-3-Clause license to the repository.

Regarding attribution on the frontend, I haven't seen that in any Streamlit app I've come across, but let me check in the Streamlit forums to be sure. Thanks again!

**Reply by stefan6419846** [reply]
Thanks for clarifying the repository license. By the way: We usually refer to *pypdf* instead of *PyPDF*.

**Reply by SiddhantSadangi** [reply]
Noted and updated 📝 
Thanks!
